swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -139,6 +139,38 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
         resourceListJSON in
         completionHandler?(resourceListJSON?.data(using: .utf8))
       }
+    case .clearLogs:
+      clearLogs(completionHandler)
+    case .getLogFolderSize:
+      getLogFolderSize(completionHandler)
+    }
+  }
+
+  func clearLogs(_ completionHandler: ((Data?) -> Void)? = nil) {
+    Task {
+      do {
+        try Log.clear(in: SharedAccess.logFolderURL)
+      } catch {
+        Log.tunnel.error("Error clearing logs: \(error)")
```

Comment:
Why doesn't this just bubble up? If you are not doing anything after the try-catch, it doesn't really make sense already catch the error here, right?

Reply:
Yeah this same thought crossed my mind. I'll refactor this a bit.

Reply:
In this particular case do note that we can't throw errors across the IPC boundary unless we use an encoding. Depending on the nature of the IPC call we're making it may not matter if they fail.

In the future I'd like to add a generic `ErrorDialog` notification in the app process that allows us to bubble actionable errors up to the user.

Reply:
In the Rust GUI client, we send a `Result<T, String>` back.

Reply:
This is a bit of a ball of yarn, so I opened https://github.com/firezone/firezone/issues/7549 to make this nicer.